### PR TITLE
Implement community board (flyer tests) [2/7]

### PR DESCRIPTION
**Pull request by isaachan100**
## Overview

Implement jest testing for FlyerRepo

## Changes Made

Create unit jest test cases for FlyerRepo

## Test Coverage

jest testing runs to completion

## Next Steps (delete if not applicable)

Next steps: make PRs 3-7 that will implement organizations, categories, and user routes

### CODE REVIEW

File: src/tests/flyer.test.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    await FlyerModel.insertMany(flyers);
+
+    const getFlyersResponse = await FlyerRepo.getFlyersAfterDate(
+      faker.date.recent(2, today.getDate() - 1).toString(),
+    ); // find flyers from after 2 days ago
+    expect(getFlyersResponse[0].title).toEqual(flyers[0].title);
+  });
+});
+
+describe('searchFlyer tests', () => {
```
Comment (katesliang) [inline src/tests/flyer.test.ts:147]: add test for limit and a test where you expect 0 results

**Review by katesliang — CHANGES_REQUESTED** [review]
nicee! should change your PR title to `Implement Flyer tests [2/7]` to distinguish the PRs in your stack

also why are no tests for `getTrendingFlyers` and `refreshTrendingFlyers`?

File: src/tests/flyer.test.ts (hunk 1 above)
Comment (isaachan100) [reply src/tests/flyer.test.ts:147]: will do! thanks for the suggestion

File: src/tests/data/FlyerFactory.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+     * Returns a list of n number of random Flyer objects with specified
+     * fields values in newMappings
+     *
+     * @param n The number of desired random Flyer objects
+     * @param newMappings The specified values for Flyer parameters [key]
+     * @returns A Promise of the list of n number of random Flyer objects
+     */
+    const arr = await FlyerFactory.create(n);
+    return arr.map((x) => {
+      const newDoc = x;
```
[the remark this answers is not in the document]
Comment (isaachan100) [reply src/tests/data/FlyerFactory.ts:34]: unfortunately idt it does

File: src/tests/data/FlyerFactory.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    /**
+     * Returns a Flyer with random values in its instance variables
+     *
+     * @returns The Flyer object with random values in its instance variables
+     */
+    const fakeFlyer = new Flyer();
+    const exampleOrg = await OrganizationFactory.getRandomOrganization();
+
+    fakeFlyer.date = faker.date.past();
+    fakeFlyer.imageURL = faker.image.cats();
```
[the remark this answers is not in the document]
Comment (isaachan100) [reply src/tests/data/FlyerFactory.ts:52]: meow

**Review by katesliang — APPROVED** [review]
prod

**Review by snajima — DISMISSED** [review]
lgtm! can we add spaces between all the tests before we merge

**Review by katesliang — APPROVED** [review]
nice

also you don't need to re-request reviews after it's been approved

**Review by snajima — APPROVED** [review]

**Pull request merged**